src/app/clusters/thread-network-diagnostics-server/thread-network-diagnostics-provider.cpp

```diff
@@ -682,7 +682,15 @@ CHIP_ERROR WriteThreadNetworkDiagnosticAttributeToTlv(AttributeId attributeId, a
         }
         else
         {
-            err = encoder.Encode(*(uint64_t *) extAddress->m8);
+            // This attribute's value is composed by taking the 8 octets of the extended address EUI-64 and
+            // treating them as a big-endian integer.
+            uint64_t extAddrBigEndian = 0;
+            for (size_t index = 0; index < sizeof(extAddress->m8); ++index)
+            {
+                extAddrBigEndian = extAddrBigEndian << 8;
+                extAddrBigEndian += extAddress->m8[index];
```

Comment:
Please just use our existing BigEndian reader bits: include CHIPEncoding.h and:
```
   encoder.Encode(Encoding::BigEndian::Get64(extAddress->m8));
```

maybe with a `static_assert(sizeof(extAddress->m8) == 8, "Unexpected buffer size")`.

Reply:
Done